app/models/piece.rb

```diff
@@ -42,8 +42,15 @@ def blocked?(x, y)
     end
 
     piece_path.each do |square|
-      return true if game.is_occupied?(square[0], square[1])
+      return true if game.square_occupied?(square[0], square[1])
     end
     false
   end
+
+  def move_to!(x, y) # move and capture
+    @game = game
```

Comment:
So, since you're already inside the piece class, you have access to the current piece and the game it belongs to, so you don't need to assign `game` to `@game` to use it in here.  It can just be `game`.